### ISSUE TITLE
Reconnecting after power outage

### ISSUE THREAD

**Issue by scargill** [hardillb/node-red-contrib-googlehome-web#77]
Hi

I have set up your node set to control 4 items, my office lights (contolled by NR), garden lights etc, all controlled by Node-Red.

I set it all up and all worked perfectly, my Google Home APP has controls for aach and I can talk to the phone and the relevant lights turn on and off on verbal (or button) command - I hd this all perfect for a couple of days - HOWEVER - today we had a power and internet outage. Everything has been back on for several hours, all reconnected.

BUT -  my NRC-googlehome nodes are now showing "reconnecting" constantly in the Node-Red main flow panel... pressing a button on my phone in Google Home for the office lights (assuming they are on) shows them going to OFF then straight back ON while never altering the actual lights  - and speech "Hey Google - turn off the office lights" - produces "sorry it looks like the office lights isnt available right now" -  strangely, my (unrelated) Alexa routine nodes are showing - when given some injected speech to send to an Alexa device - "Error: Timeout at ClientR..." yet Node-Red-dashboard in the house on the phone - controls the office lights - and if I'm not at home, the same lights in Node-Red dashboard (I use tailscale on the RPi running NR and on my phone) again works - what could have gone wrong today with the connection with your nodes?

I'm drawing a blank here. Any ideas welcome...

**Comment by hardillb** [comment]
Are you sure that when things restarted that a second instance of Node-RED has been restarted that was previously disbabled? You can only have 1 instance of the service running (that means a single config node, as the username is used a MQTT client id).

If not then you need to check if you can ping `googlehome.hardill.me.uk` (both IPv4 and IPv6 if your network supports both) from the machine running Node-RED.

Is there anything in the Node-RED console output, rather than just the Editor/Sidebar?

**Comment by scargill** [comment]
Hi - thank you for that. Ok, so I SSH'd to the RPi4 - and typed node-red-stop - indeed it stopped i.e. no running instance - so I hit node-red-start and it started - indicating one NR running.  I then went into the Node-Red screen on my PC to look at the flows.. and sure enough - still your nodes "disconnected - reconnecting" - and the node-red-speech nodes for Alexa (not yours) which work on an inject still said "open 192.168.1.19:3456 in your browser"..... when I did that, I goty "Proxy-Error: Error: getaddrinfo EAI_AGAIN www.amazon.com" - any time I've done that in the past - I would get an authentication page and Amazon starts up perfectly - but not this time. (again .19 is my RPi4 with Node-Red, MQTT etc on it).

As far as the NR logs showrf - SSHing into the RPi, I saw in the  updating log... "[error] [google-home-conf:28cb9ee1ac6a691c] Error: getaddrinfo EAI_AGAIN googlehome.hardill.me.uk"

I tried the ping... using ssh as usual on my PC to the RPI...

pi@mainpi:~:12:03[130]> ping googlehome.hardill.me.uk
ping: googlehome.hardill.me.uk: Temporary failure in name resolution
pi@mainpi:~:12:03[2]>

I had abcolutely no idea why a ping should fail.

From my Windows PC - ping..

C:\Users\pete>ping googlehome.hardill.me.uk
Pinging googlehome.hardill.me.uk [34.241.226.122] with 32 bytes of data:
Reply from 34.241.226.122: bytes=32 time=88ms TTL=45

from the RPi...

pi@mainpi:~:12:05[2]> ping googlehome.hardill.me.uk
ping: googlehome.hardill.me.uk: Temporary failure in name resolution
pi@mainpi:~:12:07[2]>

It's dawned on me that the RPi had an issue, no idea what...  I stopped the Pi and pulled the SD out, leaving it to run on an SSD - backup from a while ago....oh dear, I've been remiss in making backups (rpi-clone - takes no time)
and yes it would ping..
pi@mainpi:~:12:16[2]> ping googlehome.hardill.me.uk
PING googlehome.hardill.me.uk (34.241.226.122) 56(84) bytes of data.
64 bytes from ec2-34-241-226-122.eu-west-1.compute.amazonaws.com (34.241.226.122): icmp_seq=1 ttl=45 time=74.5 ms

SO...... I went back to the original SD and LO -  yours works - and thanls so much... 
Amazon on the other hand as usual wanted me to put in 192.168.1.19:3456 into my browser - now taking me to a valid page - but when I signed in - it started aking security questions andI ould not get past that...

Another good reason to migrate to Google. I'm going to buy a Google Hub Mini on Monday and try to get speech to come out of that,
![image](https://github.com/hardillb/node-red-contrib-googlehome-web/assets/1889411/31cf3638-6ad5-4e8a-81e5-67194b56a3c0)
